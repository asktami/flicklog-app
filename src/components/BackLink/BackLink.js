import React from 'react';
import { useHistory } from 'react-router-dom';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const BackLink = ({ label, text }) => {
	let history = useHistory();

	const handleClickGoBack = () => {
		history.goBack();
	};
	return (
		<>
			<button
				aria-expanded="false"
				aria-label="back-button"
				className="btn-as-link"
				onClick={handleClickGoBack}
			>
				{label === 'icon' ? (
					<>
						<FontAwesomeIcon icon={['fas', 'chevron-circle-left']} size="1x" />{' '}
						{text}
					</>
				) : (
					text
				)}
			</button>
			<br />
			<br />
		</>
	);
};

export default BackLink;
