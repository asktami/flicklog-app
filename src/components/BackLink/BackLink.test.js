import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));


import BackLink from './BackLink';

describe(`BackLink component`, () => {
	it('renders a BackLink by default', () => {
		const wrapper = shallow(<BackLink />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
