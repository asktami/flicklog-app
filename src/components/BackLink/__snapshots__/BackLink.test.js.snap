// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackLink component renders a BackLink by default 1`] = `
<Fragment>
  <button
    aria-expanded="false"
    aria-label="back-button"
    className="btn-as-link"
    onClick={[Function]}
  />
  <br />
  <br />
</Fragment>
`;
